packages/database/migrate_data/20221010203922_copy_existing_coupon_id_to_redeemed_for_purchase.ts

```diff
@@ -0,0 +1,37 @@
+// const {prisma} = require('../src/client')
+import {prisma} from '../src/client'
+
+const copyOverCouponIdToRedeemedCouponId = async () => {
+  const purchaseCount = await prisma.purchase.count()
+  const purchasesThatNeedUpdating = await prisma.purchase.findMany({
+    where: {redeemedCouponId: null, couponId: {not: null}},
+  })
+
+  console.log(
+    `Need to update ${purchasesThatNeedUpdating.length} of ${purchaseCount} purchases.`,
+  )
+
+  purchasesThatNeedUpdating.forEach(async (purchase) => {
+    await prisma.purchase.update({
+      where: {
+        id: purchase.id,
+      },
+      data: {
+        redeemedCouponId: purchase.couponId,
+      },
+    })
+  })
+
+  const afterPurchasesThatNeedUpdating = await prisma.purchase.findMany({
+    where: {redeemedCouponId: null, couponId: {not: null}},
+  })
+
+  const numberUpdated =
+    purchasesThatNeedUpdating.length - afterPurchasesThatNeedUpdating.length
+  console.log(`Successfully updated ${numberUpdated} purchases`)
+}
+
+// copyOverCouponIdToRedeemedCouponId()
+console.log({database_url: process.env.DATABASE_URL})
```

Comment:
Possibly my brain is fried from a long day, but I'm struggling to
understand where the node environment is being derived from when I
execute this script. No matter what directory I execute this from,
`ts-node` has the impression that `DATABASE_URL` is set to the URL
defined in TA's `.env` file. What am I missing? How do I make the
environment variables specific to the project this script is being run
in?

Reply:
TA is the only product with purchases so I don't think this needs to be codified in the package scope

I'd also hesitate to make it appear in a format that makes it look like it should/is a "rails style migration" and not just a one off since part of the purpose here is to make these changes before more than just TA need them. flexible on this, just want to throw it out there

Reply:
> TA is the only product with purchases so I don't think this needs to be codified in the package scope

I was hoping to answer the question of "how do I migrate data across app databases?" sooner rather than later. It won't be long before we have data in both that needs to be massaged.

> I'd also hesitate to make it appear in a format that makes it look like it should/is a "rails style migration"

I went overboard with the filename. It is intended to be a one-off migration.